rocksdb.go

```diff
@@ -50,7 +51,10 @@ func NewRocksDBWithOptions(name string, dir string, opts *gorocksdb.Options) (*R
 	dbPath := filepath.Join(dir, name+".db")
 	db, err := gorocksdb.OpenDb(opts, dbPath)
 	if err != nil {
-		return nil, err
+		err = os.MkdirAll(dbPath, 0755)
```

Comment:
`gorocksdb.OpenDb` calls the C library's `rocksdb_open` which already creates the directory if it doesn't exist ([ref](https://github.com/facebook/rocksdb/blob/736a7b5433025a84efc01836c145ee5afa5cffb1/db/db_impl/db_impl_open.cc#L1685)). So this is harmless, but should not be necessary for RocksDB.

Reply:
not for state sync snapshots.

I am not sure why that is the case, but it is the case.

Reply:
I suppose there is no harm in creating the directory. But isn't this backwards? If we don't create the directory till after the `OpenDb` call has already failed, the code below will fail anyway.

I'm guessing this probably wants to be something like

```go
if err := os.MkdirAll(dbPath, 0700); err != nil {
   return nil, fmt.Errorf("creating database path: %w", err)
}
db, err := gorocksdb.OpenDb(opts, dbPath)
if err != nil {
   return nil, fmt.Errorf("opening database: %w", err)
}
```

or words to that effect.

Reply:
I don't know why this works.  Let me show ya what I have......